py-client/client_api.py

```diff
@@ -46,343 +61,390 @@ class AIAAClient:
+        lines.append('')
+
+        body = bytearray()
+        for l in lines:
+            body.extend(l if isinstance(l, bytes) else l.encode('utf-8'))
+            body.extend(b'\r\n')
+
+        content_type = 'multipart/form-data; boundary=%s' % limit
+        return content_type, body
+
+    @staticmethod
+    def get_content_type(filename):
+        return mimetypes.guess_type(filename)[0] or 'application/octet-stream'
+
+    @staticmethod
+    def parse_multipart(fp, headers):
+        logger = logging.getLogger(__name__)
+        fs = cgi.FieldStorage(
+            fp=fp,
+            environ={'REQUEST_METHOD': 'POST'},
+            headers=headers,
+            keep_blank_values=True
+        )
+        form = {}
+        files = {}
+        for f in fs.list:
+            logger.debug('FILE-NAME: {}; NAME: {}; SIZE: {}'.format(f.filename, f.name, len(f.value)))
+            if f.filename:
+                files[f.filename] = f.value
+            else:
+                form[f.name] = f.value
+        return form, files
 
-    response_info = r.headers['Content-Type']
-    print(response_info)
-
-    files = []
-    if response_info.find("json") != -1:
-        # save json
-        result_path = result_prefix + '.json'
-        with open(result_path, 'wb') as f:
-            f.write(r.content)
-            print('Write returned json to', result_path)
-            files.append(result_path)
-    elif response_info.find("multipart") != -1:
-        # parse multipart data
-        multipart_data = decoder.MultipartDecoder.from_response(r)
-        for part in multipart_data.parts:
-            print(part.headers)
-
-            header = part.headers[b'Content-Disposition']
-
-            header = header.decode('utf-8')
-            header_info = header.split(";")
-
-            if header_info[0] != 'form-data':
-                continue
-
-            name_header = re.findall("name=\"(.+)\"|$", header_info[1])[0]
-
-            result_path = ""
-
-            if name_header == 'points':
-                result_path = result_prefix + '_points.txt'
-            elif name_header == 'results':
-                result_path = result_prefix + '.json'
-            elif name_header == "file":
-                filename_header = re.findall("name=\"(.+)\"|$", header_info[2])[0]
-                result_path = result_prefix + "_" + filename_header
-                if filename_header.find(".nii.gz") != -1:
-                    result_path = result_prefix + "_aas_result.nii.gz"
-                if filename_header.find(".png") != -1:
-                    result_path = result_prefix + "_aas_result.png"
-
-            if result_path:
-                print("Write returned file to ", result_path)
-                with open(result_path, 'wb') as f:
-                    f.write(part.content)
-                    files.append(result_path)
-
-    return files
+    # noinspection PyUnresolvedReferences
+    @staticmethod
+    def urllib_quote_plus(s):
```

Comment:
Should we also put this part in the very front, like what you have for httplib import.
Instead of wrapping this as a function in AIAAUtils?

Reply:
good point..